File: libgdx_ninja_runner/core/src/com/nopalsoft/ninjarunner/objects/Wall.java
package com.nopalsoft.ninjarunner.objects;

import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.utils.Pool.Poolable;

public class Wall implements Poolable {
    public final static int STATE_NORMAL = 0;
    public final static int STATE_DESTROY = 1;
    public static final float WIDTH = .98f;
    public static final float HEIGHT = 4.30f;
    public final Vector2 position;
    public int state;
    public float stateTime;

    public Wall() {
        position = new Vector2();
    }

    public void init(float x, float y) {
        position.set(x, y);
        state = STATE_NORMAL;
        stateTime = 0;

    }

    public void update(float delta) {
        stateTime += delta;
    }

    public void setDestroy() {
        if (state == STATE_NORMAL) {
            state = STATE_DESTROY;
            stateTime = 0;
        }
    }

    @Override
    public void reset() {
    }
}

File: libgdx_ninja_runner/core/src/com/nopalsoft/ninjarunner/objects/Wall.kt
package com.nopalsoft.ninjarunner.objects

import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.utils.Pool.Poolable

class Wall : Poolable {

    @JvmField
    val position = Vector2()
    @JvmField
    var state = 0
    var stateTime = 0f

    fun initializeWall(x: Float, y: Float) {
        position.set(x, y)
        state = STATE_NORMAL
        stateTime = 0f
    }

    fun update(delta: Float) {
        stateTime += delta
    }

    fun setDestroy() {
        if (state == STATE_NORMAL) {
            state = STATE_DESTROY
            stateTime = 0f
        }
    }

    override fun reset() {
        // Nothing is going on in here.
    }

    companion object {
        const val STATE_NORMAL = 0
        const val STATE_DESTROY = 1
        const val WIDTH = .98f
        const val HEIGHT = 4.30f
    }
}